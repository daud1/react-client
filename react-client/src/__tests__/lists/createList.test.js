import React from 'react';
import {shallow, render, mount } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import CreateList from '../../components/lists/createList.component';
import moxios from 'moxios';

describe ('CreateList Form Component', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<CreateList />);
        expect(wrapper.exists(<div></div>)).toBe(true);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<CreateList />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('handles successful axios calls properly', (done) => {
        const wrapper = shallow(<CreateList />);

        wrapper.find('input[name="name"]').simulate('change', { target: { value: 'newList' } });
        wrapper.find('form').simulate('submit', { preventDefault() {} });

        moxios.stubRequest('http://localhost:5000/shoppinglists/', {
            status: 201,
            'MSG': 'List created!'
        });
        moxios.wait(() => {
            expect(wrapper.html()).toContain('List created!');
            done();
        });
    });
})