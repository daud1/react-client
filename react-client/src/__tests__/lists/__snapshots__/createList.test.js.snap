// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateList Form Component matches snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Enter list name"
      fullWidth={false}
      multiLine={false}
      name="name"
      rows={1}
      type="text"
      underlineShow={true}
    />
    <button
      className="btn btn-primary"
      type="submit"
    >
      Create List
    </button>
  </form>
</div>
`;
